pkg/controller/lifecycle/actuator.go

```diff
@@ -134,44 +134,19 @@ func (a *actuator) Reconcile(ctx context.Context, ex *extensionsv1alpha1.Extensi
 	}
 
 	// initialize SecretsManager based on Cluster object
-	var (
-		caName        = "ca-extension-shoot-oidc-service"
-		secretConfigs = []extensionssecretsmanager.SecretConfigWithOptions{
-			{
-				Config: &secretutils.CertificateSecretConfig{
-					Name:       caName,
-					CommonName: caName,
-					CertType:   secretutils.CACert,
-				},
-				Options: []secretsmanager.GenerateOption{secretsmanager.Persist()},
-			},
-			{
-				Config: &secretutils.CertificateSecretConfig{
-					Name:       constants.WebhookTLSSecretName,
-					CommonName: constants.ApplicationName,
-					DNSNames:   kutil.DNSNamesForService(constants.ApplicationName, namespace),
-					CertType:   secretutils.ServerCert,
-				},
-				// use current CA for signing server cert to prevent mismatches when dropping the old CA from the webhook
-				// config in phase Completing
-				Options: []secretsmanager.GenerateOption{secretsmanager.SignedByCA(caName, secretsmanager.UseCurrentCA)},
-			},
-		}
-	)
-
-	secretsManager, err := extensionssecretsmanager.SecretsManagerForCluster(ctx, a.logger.WithName("secretsmanager"), clock.RealClock{}, a.client, cluster, constants.SecretsManagerIdentity, secretConfigs)
+	secretsManager, err := extensionssecretsmanager.SecretsManagerForCluster(ctx, a.logger.WithName("secretsmanager"), clock.RealClock{}, a.client, cluster, secrets.ManagerIdentity, secrets.ConfigsFor(namespace))
```

Comment:
Extract a variable for `secrets.ConfigsFor(namespace)` instead of calling it twice?